server/src/com/thoughtworks/go/server/materials/MaterialUpdateService.java

```diff
@@ -185,36 +173,36 @@ public void updateMaterial(Material material) {
                         "Material update is currently running but has not shown any activity in the last " + idleTime / 60000 + " minute(s). This may be hung. Details - " + material.getLongDescription(),
                         general(scope)));
             }
+            return false;
         }
     }
 
-    private boolean isConfigMaterial(Material material) {
-        return watchList.hasConfigRepoWithFingerprint(material.getFingerprint());
-    }
-
-    private Long getMaterialUpdateInActiveTimeoutInMillis() {
-        return systemEnvironment.get(SystemEnvironment.MATERIAL_UPDATE_INACTIVE_TIMEOUT) * 60 * 1000L;
+    public void registerMaterialSources(MaterialSource materialSource) {
+        this.materialSources.add(materialSource);
     }
 
     public void onMessage(MaterialUpdateCompletedMessage message) {
         try {
             if (LOGGER.isDebugEnabled()) {
                 LOGGER.debug(format("[Material Update] Material update completed for material %s", message.getMaterial()));
             }
-            updateLastUpdateTimeForScmMaterial(message.getMaterial());
+
             Date addedOn = inProgress.remove(message.getMaterial());
             serverHealthService.removeByScope(HealthStateScope.forMaterialUpdate(message.getMaterial()));
             if (addedOn == null) {
                 LOGGER.warn(format("[Material Update] Material %s was not removed from those inProgress. This might result in it's pipelines not getting scheduled. in-progress: %s",
                         message.getMaterial(), inProgress));
             }
+
+            for (MaterialUpdateCompleteListener listener : materialUpdateCompleteListeners) {
+                listener.onMaterialUpdate(message.getMaterial());
+            }
         } finally {
             mduPerformanceLogger.completionMessageForMaterialReceived(message.trackingId(), message.getMaterial());
         }
     }
 
     public void onConfigChange(CruiseConfig newCruiseConfig) {
-        updateSchedulableMaterials(true);
         Set<HealthStateScope> materialScopes = toHealthStateScopes(newCruiseConfig.getAllUniqueMaterials());
```

Comment:
Is it ok that this calls `newCruiseConfig.getAllUniqueMaterials()`? This will presumably give dependency materials as well. Does this need to change to use `sources` or something?